src/intrinsics/ecma262/MapPrototype.js

```diff
@@ -38,7 +37,8 @@ export default function(realm: Realm, obj: ObjectValue): void {
     }
 
     // 4. Let entries be the List that is the value of M's [[MapData]] internal slot.
-    let entries = ThrowIfInternalSlotNotWritable(realm, M, "$MapData").$MapData;
+    realm.recordModifiedProperty((M: any).$MapData_binding);
```

Comment:
This makes sure that the $MapData property is tracked in the current effects and thus enter the join logic.

Note that the $MapData property does not show up anywhere but in Map specific code and in the generic join logic.